### PR TITLE
Fix FYSETC/MKS/BTT Mini 12864 NeoPixel Sanity Check (again)

### PR DESCRIPTION
**Pull request by thisiskeithb**
### Description

This started as a report of flipped Red & Green colors for a FYSETC Mini 12864 V2.1 in Discord and made me dig into related PRs.

This PR reverts #24011 and updates warnings & comments to reflect correct LCD types. This was done silently before and I misunderstood why the change was made in #23590.

Tested fix on FYSETC Mini 12864 V2.1  (thanks BryanSmithDev!) and BTT Mini 12864 V1 and colors work as they should.

### Requirements

FYSETC Mini 12864 2.1, MKS Mini 12864 V3, or BTT Mini 12864 V1

### Benefits

Red & Green colors won't be flipped.

### Related Issues

- #24011
- #23590

### CODE REVIEW

**Comment by BryanSmithDev** [comment]
I can confirm this fixes the flipped colors on my FYSETC Mini 12864 V2.1.

**Comment by thinkyhead** [comment]
Are we sure all LCDs made by "MKS" and "BTT" are going to fit in this category?

**Comment by thinkyhead** [comment]
Just changed the label so it's clear that the "mini panel" applies, not all LCDs.

**Comment by thinkyhead** [comment]
How did we previously get the wrong idea that these panels required `NEO_GRB`? Are there some panels that do, or was this just a result of some confusion?

File: Marlin/src/inc/SanityCheck.h [hunk 1]
```diff
@@ -2443,23 +2443,21 @@ static_assert(Y_MAX_LENGTH >= Y_BED_SIZE, "Movement bounds (Y_MIN_POS, Y_MAX_POS
...
-    #error "Your FYSETC Mini Panel requires NEOPIXEL_PIXELS >= 3."
+    #error "Your FYSETC/MKS/BTT Mini Panel requires NEOPIXEL_PIXELS >= 3."
   #endif
   #if FAUX_RGB
-    #undef NEO_GRB
-    #undef FAUX_RGB
+    #undef NEO_RGB
   #endif
-#elif EITHER(FYSETC_MINI_12864_1_2, FYSETC_MINI_12864_2_0) && !DISABLED(RGB_LED)
+#elif EITHER(FYSETC_MINI_12864_1_2, FYSETC_MINI_12864_2_0) && DISABLED(RGB_LED)
```
Comment (thinkyhead) [inline Marlin/src/inc/SanityCheck.h:2460]: I think this is correct, based on the #error message it emits.

**Comment by thisiskeithb** [comment]
> How did we previously get the wrong idea that these panels required `NEO_GRB`? Are there some panels that do, or was this just a result of some confusion?

Marlin was silently setting it to RGB and vendor documentation was not consistent. This made old configs appear correct.

Soooo, just general confusion.

File: Marlin/src/inc/SanityCheck.h (hunk 1 above)
Comment (thisiskeithb) [reply Marlin/src/inc/SanityCheck.h:2460]: I don't think I changed that...

I also don't have these versions of Fysetc's Mini 12864 LCDs, but like you said, it seems to match the error message now.

**Comment by Sonicboom247** [comment]
i was really hoping this was the reason i am not able to get the Neopixel colors to work on an MKS mini 12864 V3 with either an MKS GEN L V1 or Artillery Ruby Board., guess it has more to do with the serial pin..

**Pull request merged**